### DISCUSSION TITLE
allow webgpu outside web

### DISCUSSION THREAD

**Discussion by octanejohn** [godotengine/godot-proposals#10292] [Platforms]
hardware that are lower than shader model 6.0/vulkan 1.0 or higher than opengl 3.3/gles 3.0/dx10 are stuck in compatibility renderer

use webgpu core(webgpu compat mode?) like what angle is doing to fill this gap

releated
https://github.com/gpuweb/gpuweb/issues/4266
https://github.com/gpuweb/gpuweb/issues/4721

edit: answered in contributors chat

**Comment by Calinou** [comment]
> hardware that are lower than shader model 6.0/vulkan 1.0 or higher than opengl 3.3/gles 3.0/dx10 are stuck in compatibility renderer

Can you point out to specific GPU models where this would be the case? I can't recall any personally.

AMD's GCN 1.0 GPUs (e.g. Radeon HD 7970) are too limited to even support the Mobile rendering method, as they don't have any Vulkan support and only have limited Direct3D 12 support. They can't run most AAA games in Direct3D 12 mode for that reason, only their D3D11 fallback if one exists.

NVIDIA's Kepler GPUs (e.g. GeForce GTX 680) only have limited Vulkan support on desktop, and no Vulkan support at all in laptops. They have a limited form of Direct3D 12 support available but again, I wouldn't expect it to run modern Direct3D 12 applications.

GPUs that are older than those (or integrated GPUs from that era) are too slow to benefit from the features exposed by the Forward+ and Mobile rendering methods, and should use Compatibility purely for performance reasons anyway. This is one of the reasons why Compatibility is designed to be as fast as possible on low-end devices after all.

**Reply by octanejohn** [reply]
most games are min dx12 sm 5.1 or feature level 11.0 and godot is min dx12 sm 6.0 so is missing the dx12 feature level 11.0 hardware, this year and next games will be defaulting to min dx12 sm 6.0

alot of android and linux phones/pcs dont/wont support vulkan and mesa devs said they wont fix

thats why i said use webgpu with fallback to opengl 4.3/gles 3.1/dx11 instead of writing an opengl version for rendering device

is not about perfomance is not able to access the rendering device features
